### DISCUSSION TITLE
Two routes to the same component?

### DISCUSSION THREAD

**Discussion by Josrph** [remix-run/remix#4905] [Q&A] [answered]
I have two different routes that point to the same component:

```
/post/pageSize/pageIdx => <Posts />
/postType/postTypeId/post/pageSize/pageIdx  => <Posts />
```

Here's what the paths look like when the dynamic parts are populated:

```
/post/1/1
/category/1/post/1/1
```

Is it possible in Remix.run and How?

**Comment by sergiodxa** [comment, marked as answer]
Move the component to another file, and import it and use it in both routes.

```
app/components/posts.tsx
app/post.$pageSize.$pageIndex.tsx <- use posts.tsx here
app/category.$postTypeId.post.$pageSize.$pageIndex.tsx <- use posts.tsx here
```

Another option is to use the new optional segments.

```
app/(category.$postTypeId).post.$pageSize.$pageIndex.tsx
```

With that, category and postTypeid are optional.

> **Note** I haven't tried doing `(category.$postTypeId)` with the `.` delimited to mark two levels of optional segments.

**Reply by alexmuro** [reply]
Came here with this exact question, had to update to 1.9.0 to get the optional routes update, and its the perfect solution.
